README.md
[![Angular Logo](https://www.vectorlogo.zone/logos/angular/angular-icon.svg)](https://angular.io/) [![Electron Logo](https://www.vectorlogo.zone/logos/electronjs/electronjs-icon.svg)](https://electronjs.org/)

![Maintained][maintained-badge]
[Make a pull request][prs]
[![License][license-badge]](LICENSE.md)

[Linux Build][linux-build]
[MacOS Build][macos-build]
[Windows Build][windows-build]

[Watch on GitHub][github-watch]
[Star on GitHub][github-star]
[Tweet][twitter]

# Introduction

Bootstrap and package your project with Angular 14 and Electron 21 (Typescript + SASS + Hot Reload) for creating Desktop applications.

Currently runs with:

- Angular v14.2.6
- Electron v21.1.1

With this sample, you can:

- Run your app in a local development environment with Electron & Hot reload
- Run your app in a production environment
- Package your app into an executable file for Linux, Windows & Mac

/!\ Hot reload only pertains to the renderer process. The main electron process is not able to be hot reloaded, only restarted.

/!\ Angular CLI & Electron Builder needs Node 14 or later to work correctly.

## Getting Started

*Clone this repository locally:*

```bash
git clone https://github.com/maximegris/angular-electron.git
```

*Install dependencies with npm (used by Electron renderer process):*

```bash
npm install
```

There is an issue with `yarn` and `node_modules` when the application is built by the packager. Please use `npm` as dependencies manager.

If you want to generate Angular components with Angular-cli , you **MUST** install `@angular/cli` in npm global context.
Please follow [Angular-cli documentation](https://github.com/angular/angular-cli) if you had installed a previous version of `angular-cli`.

```bash
npm install -g @angular/cli
```

*Install NodeJS dependencies with npm (used by Electron main process):*

```bash
cd app/
npm install
```

Why two package.json ? This project follow [Electron Builder two package.json structure](https://www.electron.build/tutorials/two-package-structure) in order to optimize final bundle and be still able to use Angular `ng add` feature.

## To build for development

- **in a terminal window** -> npm start

Voila! You can use your Angular + Electron app in a local development environment with hot reload!

The application code is managed by `app/main.ts`. In this sample, the app runs with a simple Angular App (http://localhost:4200), and an Electron window.
The Angular component contains an example of Electron and NodeJS native lib import.
You can disable "Developer Tools" by commenting `win.webContents.openDevTools();` in `app/main.ts`.

## Project structure

| Folder | Description                                      |
| ------ | ------------------------------------------------ |
| app    | Electron main process folder (NodeJS)            |
| src    | Electron renderer process folder (Web / Angular) |

## How to import 3rd party libraries

This sample project runs in both modes (web and electron). To make this work, **you have to import your dependencies the right way**. \

There are two kind of 3rd party libraries :

- NodeJS's one - Uses NodeJS core module (crypto, fs, util...)
  - I suggest you add this kind of 3rd party library in `dependencies` of both `app/package.json` and `package.json (root folder)` in order to make it work in both Electron's Main process (app folder) and Electron's Renderer process (src folder).

Please check `providers/electron.service.ts` to watch how conditional import of libraries has to be done when using NodeJS / 3rd party libraries in renderer context (i.e. Angular).

- Web's one (like bootstrap, material, tailwind...)
  - It have to be added in `dependencies` of `package.json  (root folder)`

## Add a dependency with ng-add

You may encounter some difficulties with `ng-add` because this project doesn't use the defaults `@angular-builders`.
For example you can find [here](HOW_TO.md) how to install Angular-Material with `ng-add`.

## Browser mode

Maybe you only want to execute the application in the browser with hot reload? Just run `npm run ng:serve:web`.

## Included Commands

| Command                    | Description                                                                                           |
| -------------------------- | ----------------------------------------------------------------------------------------------------- |
| `npm run ng:serve`       | Execute the app in the web browser (DEV mode)                                                         |
| `npm run web:build`      | Build the app that can be used directly in the web browser. Your built files are in the /dist folder. |
| `npm run electron:local` | Builds your application and start electron locally                                                    |
| `npm run electron:build` | Builds your application and creates an app consumable based on your operating system                  |

**Your application is optimised. Only /dist folder and NodeJS dependencies are included in the final bundle.**

**FOR WINDOWS EXE BUILD:**

1. At first run the command 'npm run electron:build'
2. Then check the release forlder.

## You want to use a specific lib (like rxjs) in electron main thread ?

YES! You can do it! Just by importing your library in npm dependencies section of `app/package.json` with `npm install --save XXXXX`.
It will be loaded by electron during build phase and added to your final bundle.
Then use your library by importing it in `app/main.ts` file. Quite simple, isn't it?

## E2E Testing

E2E Test scripts can be found in `e2e` folder.

| Command         | Description              |
| --------------- | ------------------------ |
| `npm run e2e` | Execute end to end tests |

Note: To make it work behind a proxy, you can add this proxy exception in your terminal
`export {no_proxy,NO_PROXY}="127.0.0.1,localhost"`

## Debug with VsCode

[VsCode](https://code.visualstudio.com/) debug configuration is available! In order to use it, you need the extension [Debugger for Chrome](https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome).

Then set some breakpoints in your application's source code.

Finally from VsCode press **Ctrl+Shift+D** and select **Application Debug** and press **F5**.

Please note that Hot reload is only available in Renderer process.

## Branch & Packages version

- Angular 4 & Electron 1 : Branch [angular4](https://github.com/maximegris/angular-electron/tree/angular4)
- Angular 5 & Electron 1 : Branch [angular5](https://github.com/maximegris/angular-electron/tree/angular5)
- Angular 6 & Electron 3 : Branch [angular6](https://github.com/maximegris/angular-electron/tree/angular6)
- Angular 7 & Electron 3 : Branch [angular7](https://github.com/maximegris/angular-electron/tree/angular7)
- Angular 8 & Electron 7 : Branch [angular8](https://github.com/maximegris/angular-electron/tree/angular8)
- Angular 9 & Electron 7 : Branch [angular9](https://github.com/maximegris/angular-electron/tree/angular9)
- Angular 10 & Electron 9 : Branch [angular10](https://github.com/maximegris/angular-electron/tree/angular10)
- Angular 11 & Electron 12 : Branch [angular11](https://github.com/maximegris/angular-electron/tree/angular11)
- Angular 12 & Electron 16 : Branch [angular12](https://github.com/maximegris/angular-electron/tree/angular12)
- Angular 13 & Electron 18 : Branch [angular13](https://github.com/maximegris/angular-electron/tree/angular13)
- Angular 14 & Electron 21 : (main)

[maintained-badge]: https://img.shields.io/badge/maintained-yes-brightgreen
[license-badge]: https://img.shields.io/badge/license-MIT-blue.svg
[license]: https://github.com/maximegris/angular-electron/blob/main/LICENSE.md
[prs-badge]: https://img.shields.io/badge/PRs-welcome-red.svg
[prs]: http://makeapullrequest.com
[linux-build-badge]: https://github.com/maximegris/angular-electron/workflows/Linux%20Build/badge.svg
[linux-build]: https://github.com/maximegris/angular-electron/actions?query=workflow%3A%22Linux+Build%22
[macos-build-badge]: https://github.com/maximegris/angular-electron/workflows/MacOS%20Build/badge.svg
[macos-build]: https://github.com/maximegris/angular-electron/actions?query=workflow%3A%22MacOS+Build%22
[windows-build-badge]: https://github.com/maximegris/angular-electron/workflows/Windows%20Build/badge.svg
[windows-build]: https://github.com/maximegris/angular-electron/actions?query=workflow%3A%22Windows+Build%22
[github-watch-badge]: https://img.shields.io/github/watchers/maximegris/angular-electron.svg?style=social
[github-watch]: https://github.com/maximegris/angular-electron/watchers
[github-star-badge]: https://img.shields.io/github/stars/maximegris/angular-electron.svg?style=social
[github-star]: https://github.com/maximegris/angular-electron/stargazers
[twitter]: https://twitter.com/intent/tweet?text=Check%20out%20angular-electron!%20https://github.com/maximegris/angular-electron%20%F0%9F%91%8D
[twitter-badge]: https://img.shields.io/twitter/url/https/github.com/maximegris/angular-electron.svg?style=social


CHANGELOG.md
## 11.2.0 (2022-10-16)

* misc/ Upgrade to Electron 21 ([ed23e89](https://github.com/maximegris/angular-electron/commit/ed23e89))



## 11.1.0 (2022-08-12)

* [Bumped Version] 11.1.0 ([a3eeb6a](https://github.com/maximegris/angular-electron/commit/a3eeb6a))
* Bump actions/cache from 2 to 3 ([85b8624](https://github.com/maximegris/angular-electron/commit/85b8624))
* Bump actions/checkout from 2 to 3 ([0650214](https://github.com/maximegris/angular-electron/commit/0650214))
* Bump actions/setup-node from 1 to 3 ([f9a55b6](https://github.com/maximegris/angular-electron/commit/f9a55b6))
* refactor/ build electron archive as asar ([e1da5b5](https://github.com/maximegris/angular-electron/commit/e1da5b5))
* refactor/ replace deprecated url.format with URL WHATWG API ([c491c48](https://github.com/maximegris/angular-electron/commit/c491c48))



## 11.0.0 (2022-07-20)

* [Bumped Version] 11.0.0 ([91a8574](https://github.com/maximegris/angular-electron/commit/91a8574))
* misc/ Upgrade to Angular 14 & Electron 19 ([dbacec7](https://github.com/maximegris/angular-electron/commit/dbacec7))
* chore: Included githubactions in the dependabot config ([36f2a66](https://github.com/maximegris/angular-electron/commit/36f2a66))
* chore: Set permissions for GitHub actions ([ca54dda](https://github.com/maximegris/angular-electron/commit/ca54dda))



## <small>10.5.2 (2022-04-23)</small>

* [Bumped Version] 10.5.2 ([c26be0a](https://github.com/maximegris/angular-electron/commit/c26be0a))
* Bump plist from 3.0.4 to 3.0.5 ([bedaba3](https://github.com/maximegris/angular-electron/commit/bedaba3))
* fix/ macos action ([f1d89b1](https://github.com/maximegris/angular-electron/commit/f1d89b1))
* misc/ add electron-reloader ([c885a24](https://github.com/maximegris/angular-electron/commit/c885a24))



## <small>10.5.1 (2022-04-02)</small>

* [Bumped Version] 10.5.1 ([ad23ace](https://github.com/maximegris/angular-electron/commit/ad23ace))
* fix/ macos action ([ff71c32](https://github.com/maximegris/angular-electron/commit/ff71c32))
* fix/ remove demo.gif ([ccd468d](https://github.com/maximegris/angular-electron/commit/ccd468d))
* migrate to main branch ([2c7cd98](https://github.com/maximegris/angular-electron/commit/2c7cd98))
* misc/ Upgrade to Electron 18 ([7251676](https://github.com/maximegris/angular-electron/commit/7251676))
* Set theme jekyll-theme-cayman ([1f0ba34](https://github.com/maximegris/angular-electron/commit/1f0ba34))
* fix: github action ([d6b5784](https://github.com/maximegris/angular-electron/commit/d6b5784))



## 10.5.0 (2022-02-24)

* [Bumped Version] 10.5.0 ([92fe8ed](https://github.com/maximegris/angular-electron/commit/92fe8ed))
* Bump follow-redirects from 1.14.5 to 1.14.7 ([606d54d](https://github.com/maximegris/angular-electron/commit/606d54d))
* Bump log4js from 6.3.0 to 6.4.0 ([27adf26](https://github.com/maximegris/angular-electron/commit/27adf26))
* Bump nanoid from 3.1.30 to 3.2.0 ([e05d98b](https://github.com/maximegris/angular-electron/commit/e05d98b))
* fix/ global error in console when building web app only ([58424a2](https://github.com/maximegris/angular-electron/commit/58424a2))
* misc/ github actions on 14.x, 16.x, 17.x ([4fe8c2c](https://github.com/maximegris/angular-electron/commit/4fe8c2c))
* misc/ Upgrade to Electron 17 ([4964bae](https://github.com/maximegris/angular-electron/commit/4964bae))
* use local electron from devDependencies instead of npx ([a029060](https://github.com/maximegris/angular-electron/commit/a029060))



## <small>10.4.1 (2021-11-17)</small>

* [Bumped Version] 10.4.1 ([c5096fa](https://github.com/maximegris/angular-electron/commit/c5096fa))
* misc/ Upgrade eslint version ([4321d9d](https://github.com/maximegris/angular-electron/commit/4321d9d))
* misc/ Upgrade to Angular 13.0.0 ([99e9f35](https://github.com/maximegris/angular-electron/commit/99e9f35))



## 10.4.0 (2021-11-17)

* [Bumped Version] 10.4.0 ([7a4f527](https://github.com/maximegris/angular-electron/commit/7a4f527))
* misc/ upgrade to Angular 12.2.13 ([74df637](https://github.com/maximegris/angular-electron/commit/74df637))
* misc/ upgrade to Electron 16 ([65d285c](https://github.com/maximegris/angular-electron/commit/65d285c))
* misc/ upgrade wait-on dependency ([e427f26](https://github.com/maximegris/angular-electron/commit/e427f26))



## <small>10.3.1 (2021-11-17)</small>

* [Bumped Version] 10.3.1 ([ebde830](https://github.com/maximegris/angular-electron/commit/ebde830))
* fix/ TS Config : module resolution in main process ([a9d3044](https://github.com/maximegris/angular-electron/commit/a9d3044))



## 10.3.0 (2021-11-15)

* [Bumped Version] 10.3.0 ([20e9142](https://github.com/maximegris/angular-electron/commit/20e9142))
* fix/Polyfill Node.js core modules in Webpack (5+) ([e3c0d9f](https://github.com/maximegris/angular-electron/commit/e3c0d9f))
* ref/ move angularCompilerOptions to tsconfig ([e07410b](https://github.com/maximegris/angular-electron/commit/e07410b))
* remove include statement from base tsconfig.json ([aebec1a](https://github.com/maximegris/angular-electron/commit/aebec1a))
* Update README.md ([47fc3ff](https://github.com/maximegris/angular-electron/commit/47fc3ff))



## 10.2.0 (2021-11-14)

* [Bumped Version] 10.2.0 ([edb5269](https://github.com/maximegris/angular-electron/commit/edb5269))
* add missing @ on the SCSS import for Material theme fix ([dc9ee14](https://github.com/maximegris/angular-electron/commit/dc9ee14))
* bugfix/ electron builder 22.11 needs Node 14+ ([c7dbb3f](https://github.com/maximegris/angular-electron/commit/c7dbb3f))
* Bump electron from 13.1.7 to 13.3.0 ([fe0280a](https://github.com/maximegris/angular-electron/commit/fe0280a))
* remove screenshot test ([62facb1](https://github.com/maximegris/angular-electron/commit/62facb1))
* replace Spectron by Playwright ([558c646](https://github.com/maximegris/angular-electron/commit/558c646))

app/main.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
// const { spawnSync } = require('child_process');
// const pythonProcess = spawnSync("C:\Python27/python.exe", [`${__dirname}/activityLog/event-capture.py`]);
const electron_1 = require("electron");
const { BrowserWindow } = require('electron');
const path = require("path");
const fs = require("fs");
const { autoUpdater } = require('electron-updater');
const { log } = require('electron-log/main');
log.transports.file.resolvePathFn = () => path.join('D:\Trutimer\trutimer-client', 'logs/main.log');
const { spawn } = require('child_process');
// const pythonScriptPath = path.join(__dirname, '../activityLog/event_capture.py');
// let pythonProcess;
let index = 0;
let myWindow = null;
let win = null;
const args = process.argv.slice(1), serve = args.some(val => val === '--serve');
let isAppFocused = true;
let mouseData;
function createWindow() {
    try {
        singleInstance();
        const size = electron_1.screen.getPrimaryDisplay().workAreaSize;
        const fixedWidth = 900;
        const fixedHeight = 750;
        // Create the browser window.
        win = new BrowserWindow({
            x: 0,
            y: 0,
            width: fixedWidth,
            height: fixedHeight,
            minWidth: fixedWidth,
            minHeight: fixedHeight,
            webPreferences: {
                nodeIntegration: true,
                allowRunningInsecureContent: serve ? true : false,
                backgroundThrottling: false,
                contextIsolation: false, // false if you want to run e2e test with Spectron,
            },
            alwaysOnTop: true,
        });
        autoUpdater.checkForUpdatesAndNotify();
        const iconpath = path.join(__dirname, 'favicon.ico');
        win.setIcon(iconpath);
        win.setVisibleOnAllWorkspaces(true);
        const contents = win.webContents;
        if (parseInt(process.versions.electron) >= 17) {
            electron_1.desktopCapturer.getSources({ types: ['window', 'screen'] }).then((sources) => __awaiter(this, void 0, void 0, function* () {
                for (const source of sources) {
                    if (source.id.startsWith('screen')) {
                        win.webContents.send('SET_SOURCE', source.id);
                        return;
                    }
                }
            }));
        }
        if (serve) {
            const debug = require('electron-debug');
            debug();
            require('electron-reloader')(module);
            win.loadURL('http://localhost:4200');
        }
        else {
            // Path when running electron executable
            let pathIndex = './index.html';
            if (fs.existsSync(path.join(__dirname, '../dist/index.html'))) {
                // Path when running electron in local folder
                pathIndex = '../dist/index.html';
            }
            const url = new URL(path.join('file:', __dirname, pathIndex));
            win.loadURL(url.href);
        }
        win.setMenu(null);
        win.webContents.openDevTools();
        // function startPythonProcess() {
        //   pythonProcess = spawn('python', [pythonScriptPath]);
        //   let totalCounts = [];
        //   pythonProcess.stdout.on('data', (data) => {
        //     console.log(`Data from Python: ${data}`);
        //     mouseData = data.toString();
        //     // contents.send('pythonData', data.toString());
        //   });
        //   pythonProcess.on('exit', () => {
        //     console.log('Total counts of click, move, scroll:', totalCounts);
        //   });
        //   pythonProcess.stderr.on('data', (data) => {
        //     console.error(`Error from Python: ${data}`);
        //   });
        //   pythonProcess.on('close', (code) => {
        //     if (code !== 0) {
        //       console.error('Python process exited with code: ', code);
        //     }
        //   });
        // }
        // ipcMain.on('activityStart', async (event) => {
        //   startPythonProcess();
        // });
        // ipcMain.on('activityStop', async (event) => {
        //   contents.send('pythonData', mouseData);
        //   mouseData = null;
        //   if (pythonProcess) {
        //     pythonProcess.kill('SIGINT');
        //     pythonProcess = null; // Reset the process variable
        //   }
        // });
        // ipcMain.on('activityStopForAutoSave', async (event) => {
        //   contents.send('pythonData', mouseData);
        //   mouseData = null;
        //   if (pythonProcess) {
        //     pythonProcess.kill('SIGINT');
        //     pythonProcess = null; // Reset the process variable
        //   }
        //   startPythonProcess();
        // });
        //app.dock.hide();
        win.setVisibleOnAllWorkspaces(true);
        // globalShortcut.register('CommandOrControl+F', () => {
        //   win.webContents.send('on-search');
        // });
        // globalShortcut.register('CommandOrControl+R', () => {
        //   win.webContents.send('on-refresh');
        // });
        win.on("close", function (e) {
            if (win) {
                win.setAlwaysOnTop(false);
                if (contents.getURL().includes("login") || contents.getURL().includes("home")) {
                    e.preventDefault();
                    const options = {
                        buttons: ["Yes", "No"],
                        message: "Do you really want to quit?",
                    };
                    electron_1.dialog.showMessageBox(options).then((res) => {
                        if (res.response === 0) {
                            win.webContents.session.clearCache();
                            contents.send("login_app_quit");
                        }
                        else {
                            win.maximize();
                        }
                    });
                }
                else {
                    e.preventDefault();
                    const options = {
                        buttons: ["Yes", "No"],
                        message: "Do you really want to quit?",
                    };
                    electron_1.dialog.showMessageBox(options).then((res) => {
                        if (res.response === 0) {
                            win.webContents.session.clearCache();
                            contents.send("app_quit");
                        }
                        else {
                            win.maximize();
                        }
                    });
                }
            }
        });
        electron_1.app.on('browser-window-focus', () => {
            isAppFocused = true;
        });
        electron_1.app.on('browser-window-blur', () => {
            isAppFocused = false;
        });
        let contextMenu = electron_1.Menu.buildFromTemplate([
            {
                label: "Show App",
                click: function () {
                    win.show();
                },
            },
            {
                label: "Quit",
                click: function () {
                    // app.isQuiting = true
                    electron_1.app.quit();
                },
            },
        ]);
        win.on('show', () => {
            win.moveTop();
            win.focus();
            win.maximize();
        });
        win.on("minimize", function (event) {
            event.preventDefault();
        });
        win.on('hide', () => {
            win.setAlwaysOnTop(false);
        });
        return win;
    }
    catch (e) {
        fs.appendFile("./logging.txt", e, (err) => {
            if (!err) {
            }
            else {
            }
        });
    }
}
try {
    electron_1.app.commandLine.appendSwitch("disable-background-timer-throttling");
    electron_1.app.commandLine.appendSwitch("js-flags", "--max-old-space-size=8192");
    electron_1.app.on('ready', () => setTimeout(createWindow, 400));
    electron_1.app.on('window-all-closed', () => {
        if (process.platform !== 'darwin') {
            electron_1.app.quit();
        }
    });
    electron_1.app.on('activate', () => {
        if (win === null) {
            createWindow();
        }
    });
    electron_1.ipcMain.on("minimize", (event, arg) => {
        win.minimize();
    });
    electron_1.ipcMain.on('windowClose', (event) => __awaiter(void 0, void 0, void 0, function* () {
        win = null;
        if (process.platform !== 'darwin') {
            electron_1.app.quit();
        }
    }));
    electron_1.ipcMain.on('windowCloseHandler', (event) => __awaiter(void 0, void 0, void 0, function* () {
        win = null;
        if (process.platform !== 'darwin') {
            electron_1.app.quit();
        }
    }));
    electron_1.ipcMain.on("app_bring_to_front", (event, arg) => {
        //win.setAlwaysOnTop(true);
        // if (!isAppFocused) {
        //   // Bring the window to the front if the app is not focused
        //   win.focus();
        //   win.moveTop();
        // win.maximize();
        // }
        if (win.isMinimized()) {
            win.restore(); // Restore the window if it's minimized
        }
        if (!win.isVisible()) {
            win.show();
        }
        win.show(); // Show the window
        win.focus();
        win.moveTop();
        win.maximize();
    });
    electron_1.ipcMain.on('timer-stopped', () => {
        win.webContents.send('timer-stopped');
    });
    electron_1.ipcMain.on('screenshot', (event) => __awaiter(void 0, void 0, void 0, function* () {
        const screenshotdir = '../screenshoot';
        try {
            const sources = yield electron_1.desktopCapturer.getSources({
                types: ["screen"],
                thumbnailSize: { height: 720, width: 1000 },
            });
            if (sources.length >= 0) {
                const newfilename = sources[0].thumbnail.toDataURL();
                event.reply('asynchronous-reply', newfilename);
            }
        }
        catch (error) {
            console.error('Error capturing screenshot:', error);
            event.reply('asynchronous-reply', null);
        }
    }));
    electron_1.ipcMain.on('requestSystemIdleStatus', (event, idleTime) => {
        const parsedIdleTime = Number(idleTime);
        if (typeof parsedIdleTime === 'number') {
            const systemIdleState = electron_1.powerMonitor.getSystemIdleState(parsedIdleTime);
            let isSystemIdle = systemIdleState === 'idle' ? 1 : 0;
            event.reply('systemIdleStatus', isSystemIdle);
        }
        else {
            console.error('Invalid IdleTime argument:', idleTime);
        }
    });
}
catch (e) {
}
function singleInstance() {
    const gotTheLock = electron_1.app.requestSingleInstanceLock();
    if (!gotTheLock) {
        electron_1.app.quit();
    }
    else {
        electron_1.app.on("second-instance", (event, commandLine, workingDirectory) => {
            if (myWindow) {
                if (myWindow.isMinimized())
                    myWindow.restore();
                electron_1.ipcMain.on("app_bring_to_front", (event, arg) => {
                    win.setAlwaysOnTop(true, "floating");
                    win.setVisibleOnAllWorkspaces(true);
                    win.maximize();
                });
            }
        });
        electron_1.app.on("ready", () => { });
    }
}
electron_1.powerMonitor.on('suspend', () => {
    win.webContents.send('system-suspend');
});
electron_1.powerMonitor.on('lock-screen', () => {
    win.webContents.send('system-lock-screen');
});
electron_1.powerMonitor.on('resume', () => {
    win.webContents.send('system-resume');
});
electron_1.powerMonitor.on('unlock-screen', () => {
    win.webContents.send('system-unlock-screen');
});
autoUpdater.on('update-available', () => {
    log.info('update-available');
});
autoUpdater.on('checking-for-update', () => {
    log.info('checking-for-update');
});
autoUpdater.on('download-progress', () => {
    log.info('download-progress');
});
autoUpdater.on('update-downloaded', () => {
    log.info('update-downloaded');
});
//# sourceMappingURL=main.js.map